src/org/omegat/gui/exttrans/MachineTranslateTextArea.java

```diff
@@ -94,7 +94,7 @@
      *  A value should be unique and used for identify HTML element by id attribute.
      *  We use an index counter for identifier of elements.
      */
-    protected Map<MachineTranslationInfo, String> displayed = new TreeMap<>();
+    protected List<MachineTranslationInfo> displayed = new ArrayList<>();
     private int index;
```

Comment:
I don't think you need `index` at all. `index` is always equal to `displayed.length()`.

Reply:
I've removed a field `index`.  A method ` setFoundResult` get local variable `int index = displayed.size()`